function-server/main.py

```diff
@@ -44,26 +48,35 @@ def process_msg(msg, handle):
         sys.stdout = stdout
         r = handle(msg['context'], msg['payload'])
     except Exception as e:
-        err = e
-        stacktrace = traceback.format_exc().rstrip()
-        print(stacktrace, file=sys.stderr)
+        stacktrace = traceback.format_exc().splitlines()
+        err = {'type': FUNCTION_ERROR, 'message': str(e), 'stacktrace': stacktrace, 'transient': False}
```

Comment:
This is correct.

We also need to catch `ValueError` and `TypeError` which are Python equivalents (covering different facets) of Java IllegalArgumentException. We should recommend (in the README) throwing them in the function code if the input data is determined to be invalid. This is the case when the error type is `InputError`